### PR TITLE
fix(ui5-shellbar): fix menuItems cloning

### PR DESCRIPTION
**Pull request by ilhan007**
We used to just copy the text of the menuItems within the Light DOM of the ui5-shellbar, and not only the attributes of those items are missing, but also we don't update the internal items whenever the original ones change. For example: if the textContent change - it is not applied to the internal one. Adding a mutation observer keeps everything in sync. Currently we can not just forward the outside slot to the List, that is inside a Popover, that is in the static area.

Fixes: https://github.com/SAP/ui5-webcomponents/issues/1442

### CODE REVIEW

File: packages/fiori/src/ShellBar.js [hunk 1]
```diff
@@ -380,16 +380,15 @@ class ShellBar extends UI5Element {
...
-					this._menuPopoverItems = [];
-					this.menuItems.forEach(item => {
-						this._menuPopoverItems.push(item.textContent);
-					});
-					this.updateStaticAreaItemContentDensity();
-					this.menuPopover.openBy(this.shadowRoot.querySelector(".ui5-shellbar-menu-button"));
-				}
+				this._updateClonedMenuItems();
+				this.updateStaticAreaItemContentDensity();
+				this.menuPopover.openBy(this.shadowRoot.querySelector(".ui5-shellbar-menu-button"));
```
Comment (vladitasev) [inline packages/fiori/src/ShellBar.js:391]: Is it intended that this line is outside of the `if`? So we always open the popup even though it's empty?

File: packages/fiori/src/ShellBar.js [hunk 2]
```diff
@@ -641,6 +642,7 @@ class ShellBar extends UI5Element {
 	}
 
 	onExitDOM() {
+		this.menuItemsObserver.disconnect();
```
Comment (vladitasev) [inline packages/fiori/src/ShellBar.js:645]: If you disconnect it here (f.e. the dev called removeChild(shellbar), are we sure _observeMenuItems() will be called again again upon re-inserting into the DOM? I would guess so, but just to make sure.

File: packages/fiori/src/ShellBar.js [hunk 3]
```diff
@@ -819,6 +821,34 @@ class ShellBar extends UI5Element {
 		}
 	}
 
+	_updateClonedMenuItems() {
+		if (!this.menuItems.length) {
```
Comment (vladitasev) [inline packages/fiori/src/ShellBar.js:825]: If there are no more items, shouldn't we let it delete all clones too (so the next line should always run)?

Comment (ilhan007) [reply packages/fiori/src/ShellBar.js:825]: yes, I should just remove the return I think

File: packages/fiori/src/ShellBar.js (hunk 2 above)
Comment (ilhan007) [reply packages/fiori/src/ShellBar.js:645]: Should work, I read this in the DOCS:
disconnect() stops the MutationObserver instance from receiving further notifications until and unless observe() is called again

File: packages/fiori/src/ShellBar.js (hunk 1 above)
Comment (ilhan007) [reply packages/fiori/src/ShellBar.js:391]: oh, this is by mistake when I extracted the code to _updateClonedMenuItems

**Pull request merged**